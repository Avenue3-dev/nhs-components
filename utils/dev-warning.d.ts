/**
 * Print a warning when using experimental components when not in production mode.
 *
 * @param component name of the experimental component.
 */
export declare const useExperimentalWarning: (component: 'tabs' | 'character-count') => void;
/**
 * Print a warning message when not in production mode.
 *
 * @param message message to print when not it production.
 */
export declare const useDevWarning: (message: string) => void;
