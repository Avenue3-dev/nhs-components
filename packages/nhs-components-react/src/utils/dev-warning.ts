const prod = process.env.NODE_ENV === 'production'

export const experimentalWarning = (
  component: 'tabs' | 'character-count',
): void => {
  if (prod) {
    return
  }

  console.warn(`ATTENTION
  This component is tagged as experimental and therefore could change without notice. See more details about this component at https://service-manual.nhs.uk/design-system/components/${component}
  `)
}
