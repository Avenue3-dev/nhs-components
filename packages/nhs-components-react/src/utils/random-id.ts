export const generateRandomString = (length = 5): string => {
  const rand = Math.random() + 1

  return rand.toString(36).substring(2, length + 2)
}

export const generateRandomName = (prefix?: string): string => {
  const randString = generateRandomString()

  return prefix ? `${prefix}_${randString}` : randString
}

export const generateRandomId = generateRandomName
