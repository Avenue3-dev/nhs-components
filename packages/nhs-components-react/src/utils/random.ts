export const generateRandomString = (length: number = 8): string => {
  return [...Array<string>(length)]
    .map(() => Math.random().toString(36)[2])
    .join('')
}
