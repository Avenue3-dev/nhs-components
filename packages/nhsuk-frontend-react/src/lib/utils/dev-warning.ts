const prod = process.env.NODE_ENV === 'production'

/**
 * Print a warning when using experimental components when not in production mode.
 *
 * @param component name of the experimental component.
 */
export const useExperimentalWarning = (
  component: 'tabs' | 'character-count',
): void => {
  useDevWarning(`This component is tagged as experimental and therefore could change without notice.

  See more details about this component at https://service-manual.nhs.uk/design-system/components/${component}
  `)
}

/**
 * Print a warning message when not in production mode.
 *
 * @param message message to print when not it production.
 */
export const useDevWarning = (message: string): void => {
  if (prod) {
    return
  }

  console.warn(message)
}
