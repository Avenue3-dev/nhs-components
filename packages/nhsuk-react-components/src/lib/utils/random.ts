export const generateRandomString = (length = 8): string => {
  return [...Array<string>(length)]
    .map(() => Math.random().toString(36)[2])
    .join('')
}
