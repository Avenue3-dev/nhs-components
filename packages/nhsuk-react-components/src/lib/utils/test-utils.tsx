/* eslint-disable import/export */
import {
  RenderOptions,
  RenderResult,
  cleanup,
  render,
} from '@testing-library/react'
import { MemoryRouter as Router } from 'react-router-dom'
import { afterEach } from 'vitest'

afterEach(() => {
  cleanup()
})

const customRender = (
  ui: React.ReactElement,
  options: RenderOptions = {},
): RenderResult => {
  return render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => <Router>{children}</Router>,
    ...options,
  })
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
// override render export
export { customRender as render }
