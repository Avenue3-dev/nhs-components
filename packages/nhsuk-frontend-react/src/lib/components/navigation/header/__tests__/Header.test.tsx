import { render, screen } from '@/utils/test-utils'
import { describe, expect, test } from 'vitest'

import Header from '..'

describe('Header', () => {
  test.each([true, false])(
    'should match the snapshot - white equal %s',
    (white) => {
      render(
        <Header
          data-testid="header"
          orgName="Anytown Anyplace"
          orgSplit="Anywhere"
          orgDescriptor="NHS Foundation Trust"
          white={white}
        >
          <Header.Container>
            <Header.Logo />
          </Header.Container>
          <Header.Nav>
            <Header.NavItem href="#">
              Home
            </Header.NavItem>
            <Header.NavItem href="#">Get involved</Header.NavItem>
            <Header.NavItem href="#">About us</Header.NavItem>
            <Header.NavItem href="#">About us</Header.NavItem>
          </Header.Nav>
        </Header>,
      )

      expect(screen.getByTestId('header')).toMatchSnapshot()
    },
  )

  test('should match the snapshot - search', async () => {
    const container = document.createElement('div')
    container.style.width = '414px'
    container.style.height = '896px'

    render(
      <Header
        data-testid="header"
        orgName="Anytown Anyplace"
        orgSplit="Anywhere"
        orgDescriptor="NHS Foundation Trust"
      >
        <Header.Container>
          <Header.Logo />
          <Header.Content>
            <Header.Search
              data-testid="search"
              toggleProps={{ 'data-testid': 'search-toggle' }}
              closeProps={{ 'data-testid': 'search-close' }}
            />
          </Header.Content>
        </Header.Container>
        <Header.Nav>
          <Header.NavItem href="#">
            Home
          </Header.NavItem>
          <Header.NavItem href="#">Get involved</Header.NavItem>
          <Header.NavItem href="#">About us</Header.NavItem>
          <Header.NavItem href="#">About us</Header.NavItem>
        </Header.Nav>
      </Header>,
      { container: document.body.appendChild(container) },
    )

    expect(screen.getByTestId('header')).toMatchSnapshot()
  })

  test('should match the snapshot - service', () => {
    render(
      <Header data-testid="header" serviceName="Service name">
        <Header.Container>
          <Header.Logo />
          <Header.Content>
            <Header.Search />
          </Header.Content>
        </Header.Container>
      </Header>,
    )

    expect(screen.getByTestId('header')).toMatchSnapshot()
  })

  test('should match the snapshot - transactional', () => {
    render(
      <Header data-testid="header" transactional serviceName="Service name">
        <Header.Container>
          <Header.Logo />
          <Header.TransactionalLink />
        </Header.Container>
      </Header>,
    )

    expect(screen.getByTestId('header')).toMatchSnapshot()
  })
})
