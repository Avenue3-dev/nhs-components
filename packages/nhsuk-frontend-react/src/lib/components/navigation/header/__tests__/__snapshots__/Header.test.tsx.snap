// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should match the snapshot - search 1`] = `
<header
  class="nhsuk-header nhsuk-header--organisation"
  data-testid="header"
  role="banner"
>
  <div
    class="nhsuk-width-container nhsuk-header__container"
  >
    <div
      class="nhsuk-header__logo"
    >
      <a
        aria-label="Anytown Anyplace Anywhere NHS Foundation Trust homepage"
        class="nhsuk-header__link"
      >
        <svg
          class="nhsuk-logo"
          height="40"
          viewBox="0 0 40 16"
          width="100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="nhsuk-logo__background"
            d="M0 0h40v16H0z"
            fill="#005eb8"
          />
          <path
            class="nhsuk-logo__text"
            d="M3.9 1.5h4.4l2.6 9h.1l1.8-9h3.3l-2.8 13H9l-2.7-9h-.1l-1.8 9H1.1M17.3 1.5h3.6l-1 4.9h4L25 1.5h3.5l-2.7 13h-3.5l1.1-5.6h-4.1l-1.2 5.6h-3.4M37.7 4.4c-.7-.3-1.6-.6-2.9-.6-1.4 0-2.5.2-2.5 1.3 0 1.8 5.1 1.2 5.1 5.1 0 3.6-3.3 4.5-6.4 4.5-1.3 0-2.9-.3-4-.7l.8-2.7c.7.4 2.1.7 3.2.7s2.8-.2 2.8-1.5c0-2.1-5.1-1.3-5.1-5 0-3.4 2.9-4.4 5.8-4.4 1.6 0 3.1.2 4 .6"
            fill="#fff"
          />
        </svg>
        <span
          class="nhsuk-organisation-name"
        >
          Anytown Anyplace
          <span
            class="nhsuk-organisation-name-split"
          >
            Anywhere
          </span>
        </span>
        <span
          class="nhsuk-organisation-descriptor"
        >
          NHS Foundation Trust
        </span>
      </a>
    </div>
    <div
      class="nhsuk-header__content"
      id="content-header"
    >
      <div
        class="nhsuk-header__search"
      >
        <div
          class="nhsuk-header__search-wrap"
        >
          <form
            action="/search"
            class="nhsuk-header__search-form"
            method="get"
            role="search"
          >
            <label
              class="nhsuk-u-visually-hidden"
              for="search-field"
            >
              Search
            </label>
            <input
              autocomplete="off"
              class="nhsuk-search__input"
              data-testid="search"
              id="search-field"
              placeholder="Search"
              type="search"
            />
            <button
              class="nhsuk-search__submit"
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="nhsuk-icon nhsuk-icon__search"
                focusable="false"
                height="34"
                viewBox="0 0 24 24"
                width="34"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19.71 18.29l-4.11-4.1a7 7 0 1 0-1.41 1.41l4.1 4.11a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42zM5 10a5 5 0 1 1 5 5 5 5 0 0 1-5-5z"
                />
              </svg>
              <span
                class="nhsuk-u-visually-hidden"
              >
                Search
              </span>
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    class="nhsuk-navigation-container"
  >
    <nav
      aria-label="Primary navigation"
      class="nhsuk-navigation js-enabled"
      role="navigation"
    >
      <ul
        class="nhsuk-header__navigation-list"
      >
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            Home
          </a>
        </li>
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            Get involved
          </a>
        </li>
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            About us
          </a>
        </li>
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            About us
          </a>
        </li>
        <li
          class="nhsuk-mobile-menu-container nhsuk-mobile-menu-container--visible"
        >
          <button
            aria-expanded="false"
            class="nhsuk-header__menu-toggle nhsuk-header__navigation-link nhsuk-header__menu-toggle--visible"
          >
            <span
              class="nhsuk-u-visually-hidden"
            >
              Browse
            </span>
            More
            <svg
              aria-hidden="true"
              class="nhsuk-icon nhsuk-icon__chevron-down"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.5 12a1 1 0 0 1-.29.71l-5 5a1 1 0 0 1-1.42-1.42l4.3-4.29-4.3-4.29a1 1 0 0 1 1.42-1.42l5 5a1 1 0 0 1 .29.71z"
              />
            </svg>
          </button>
          <ul
            class="nhsuk-header__drop-down nhsuk-header__drop-down--hidden"
          >
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                Home
              </a>
            </li>
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                Get involved
              </a>
            </li>
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                About us
              </a>
            </li>
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                About us
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;

exports[`Header > should match the snapshot - service 1`] = `
<header
  class="nhsuk-header"
  data-testid="header"
  role="banner"
>
  <div
    class="nhsuk-width-container nhsuk-header__container"
  >
    <div
      class="nhsuk-header__logo"
    >
      <a
        aria-label="NHS homepage"
        class="nhsuk-header__link nhsuk-header__link--service"
      >
        <svg
          class="nhsuk-logo"
          height="40"
          viewBox="0 0 40 16"
          width="100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="nhsuk-logo__background"
            d="M0 0h40v16H0z"
            fill="#005eb8"
          />
          <path
            class="nhsuk-logo__text"
            d="M3.9 1.5h4.4l2.6 9h.1l1.8-9h3.3l-2.8 13H9l-2.7-9h-.1l-1.8 9H1.1M17.3 1.5h3.6l-1 4.9h4L25 1.5h3.5l-2.7 13h-3.5l1.1-5.6h-4.1l-1.2 5.6h-3.4M37.7 4.4c-.7-.3-1.6-.6-2.9-.6-1.4 0-2.5.2-2.5 1.3 0 1.8 5.1 1.2 5.1 5.1 0 3.6-3.3 4.5-6.4 4.5-1.3 0-2.9-.3-4-.7l.8-2.7c.7.4 2.1.7 3.2.7s2.8-.2 2.8-1.5c0-2.1-5.1-1.3-5.1-5 0-3.4 2.9-4.4 5.8-4.4 1.6 0 3.1.2 4 .6"
            fill="#fff"
          />
        </svg>
        <span
          class="nhsuk-header__service-name"
        >
          Service name
        </span>
      </a>
    </div>
    <div
      class="nhsuk-header__content"
      id="content-header"
    >
      <div
        class="nhsuk-header__search"
      >
        <div
          class="nhsuk-header__search-wrap"
        >
          <form
            action="/search"
            class="nhsuk-header__search-form"
            method="get"
            role="search"
          >
            <label
              class="nhsuk-u-visually-hidden"
              for="search-field"
            >
              Search
            </label>
            <input
              autocomplete="off"
              class="nhsuk-search__input"
              id="search-field"
              placeholder="Search"
              type="search"
            />
            <button
              class="nhsuk-search__submit"
              type="submit"
            >
              <svg
                aria-hidden="true"
                class="nhsuk-icon nhsuk-icon__search"
                focusable="false"
                height="34"
                viewBox="0 0 24 24"
                width="34"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19.71 18.29l-4.11-4.1a7 7 0 1 0-1.41 1.41l4.1 4.11a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42zM5 10a5 5 0 1 1 5 5 5 5 0 0 1-5-5z"
                />
              </svg>
              <span
                class="nhsuk-u-visually-hidden"
              >
                Search
              </span>
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</header>
`;

exports[`Header > should match the snapshot - transactional 1`] = `
<header
  class="nhsuk-header nhsuk-header--transactional"
  data-testid="header"
  role="banner"
>
  <div
    class="nhsuk-width-container nhsuk-header__container"
  >
    <div
      class="nhsuk-header__logo nhsuk-header__logo--only"
    >
      <a
        aria-label="NHS homepage"
        class="nhsuk-header__link nhsuk-header__link--service"
      >
        <svg
          class="nhsuk-logo"
          height="40"
          viewBox="0 0 40 16"
          width="100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="nhsuk-logo__background"
            d="M0 0h40v16H0z"
            fill="#005eb8"
          />
          <path
            class="nhsuk-logo__text"
            d="M3.9 1.5h4.4l2.6 9h.1l1.8-9h3.3l-2.8 13H9l-2.7-9h-.1l-1.8 9H1.1M17.3 1.5h3.6l-1 4.9h4L25 1.5h3.5l-2.7 13h-3.5l1.1-5.6h-4.1l-1.2 5.6h-3.4M37.7 4.4c-.7-.3-1.6-.6-2.9-.6-1.4 0-2.5.2-2.5 1.3 0 1.8 5.1 1.2 5.1 5.1 0 3.6-3.3 4.5-6.4 4.5-1.3 0-2.9-.3-4-.7l.8-2.7c.7.4 2.1.7 3.2.7s2.8-.2 2.8-1.5c0-2.1-5.1-1.3-5.1-5 0-3.4 2.9-4.4 5.8-4.4 1.6 0 3.1.2 4 .6"
            fill="#fff"
          />
        </svg>
      </a>
    </div>
    <div
      class="nhsuk-header__transactional-service-name"
    >
      <a
        class="nhsuk-header__transactional-service-name--link"
      >
        Service name
      </a>
    </div>
  </div>
</header>
`;

exports[`Header > should match the snapshot - white equal false 1`] = `
<header
  class="nhsuk-header nhsuk-header--organisation"
  data-testid="header"
  role="banner"
>
  <div
    class="nhsuk-width-container nhsuk-header__container"
  >
    <div
      class="nhsuk-header__logo"
    >
      <a
        aria-label="Anytown Anyplace Anywhere NHS Foundation Trust homepage"
        class="nhsuk-header__link"
      >
        <svg
          class="nhsuk-logo"
          height="40"
          viewBox="0 0 40 16"
          width="100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="nhsuk-logo__background"
            d="M0 0h40v16H0z"
            fill="#005eb8"
          />
          <path
            class="nhsuk-logo__text"
            d="M3.9 1.5h4.4l2.6 9h.1l1.8-9h3.3l-2.8 13H9l-2.7-9h-.1l-1.8 9H1.1M17.3 1.5h3.6l-1 4.9h4L25 1.5h3.5l-2.7 13h-3.5l1.1-5.6h-4.1l-1.2 5.6h-3.4M37.7 4.4c-.7-.3-1.6-.6-2.9-.6-1.4 0-2.5.2-2.5 1.3 0 1.8 5.1 1.2 5.1 5.1 0 3.6-3.3 4.5-6.4 4.5-1.3 0-2.9-.3-4-.7l.8-2.7c.7.4 2.1.7 3.2.7s2.8-.2 2.8-1.5c0-2.1-5.1-1.3-5.1-5 0-3.4 2.9-4.4 5.8-4.4 1.6 0 3.1.2 4 .6"
            fill="#fff"
          />
        </svg>
        <span
          class="nhsuk-organisation-name"
        >
          Anytown Anyplace
          <span
            class="nhsuk-organisation-name-split"
          >
            Anywhere
          </span>
        </span>
        <span
          class="nhsuk-organisation-descriptor"
        >
          NHS Foundation Trust
        </span>
      </a>
    </div>
  </div>
  <div
    class="nhsuk-navigation-container"
  >
    <nav
      aria-label="Primary navigation"
      class="nhsuk-navigation js-enabled"
      role="navigation"
    >
      <ul
        class="nhsuk-header__navigation-list"
      >
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            Home
          </a>
        </li>
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            Get involved
          </a>
        </li>
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            About us
          </a>
        </li>
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            About us
          </a>
        </li>
        <li
          class="nhsuk-mobile-menu-container nhsuk-mobile-menu-container--visible"
        >
          <button
            aria-expanded="false"
            class="nhsuk-header__menu-toggle nhsuk-header__navigation-link nhsuk-header__menu-toggle--visible"
          >
            <span
              class="nhsuk-u-visually-hidden"
            >
              Browse
            </span>
            More
            <svg
              aria-hidden="true"
              class="nhsuk-icon nhsuk-icon__chevron-down"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.5 12a1 1 0 0 1-.29.71l-5 5a1 1 0 0 1-1.42-1.42l4.3-4.29-4.3-4.29a1 1 0 0 1 1.42-1.42l5 5a1 1 0 0 1 .29.71z"
              />
            </svg>
          </button>
          <ul
            class="nhsuk-header__drop-down nhsuk-header__drop-down--hidden"
          >
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                Home
              </a>
            </li>
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                Get involved
              </a>
            </li>
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                About us
              </a>
            </li>
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                About us
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;

exports[`Header > should match the snapshot - white equal true 1`] = `
<header
  class="nhsuk-header nhsuk-header--organisation nhsuk-header--white"
  data-testid="header"
  role="banner"
>
  <div
    class="nhsuk-width-container nhsuk-header__container"
  >
    <div
      class="nhsuk-header__logo"
    >
      <a
        aria-label="Anytown Anyplace Anywhere NHS Foundation Trust homepage"
        class="nhsuk-header__link"
      >
        <svg
          class="nhsuk-logo"
          height="40"
          viewBox="0 0 40 16"
          width="100"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="nhsuk-logo__background"
            d="M0 0h40v16H0z"
            fill="#005eb8"
          />
          <path
            class="nhsuk-logo__text"
            d="M3.9 1.5h4.4l2.6 9h.1l1.8-9h3.3l-2.8 13H9l-2.7-9h-.1l-1.8 9H1.1M17.3 1.5h3.6l-1 4.9h4L25 1.5h3.5l-2.7 13h-3.5l1.1-5.6h-4.1l-1.2 5.6h-3.4M37.7 4.4c-.7-.3-1.6-.6-2.9-.6-1.4 0-2.5.2-2.5 1.3 0 1.8 5.1 1.2 5.1 5.1 0 3.6-3.3 4.5-6.4 4.5-1.3 0-2.9-.3-4-.7l.8-2.7c.7.4 2.1.7 3.2.7s2.8-.2 2.8-1.5c0-2.1-5.1-1.3-5.1-5 0-3.4 2.9-4.4 5.8-4.4 1.6 0 3.1.2 4 .6"
            fill="#fff"
          />
        </svg>
        <span
          class="nhsuk-organisation-name"
        >
          Anytown Anyplace
          <span
            class="nhsuk-organisation-name-split"
          >
            Anywhere
          </span>
        </span>
        <span
          class="nhsuk-organisation-descriptor"
        >
          NHS Foundation Trust
        </span>
      </a>
    </div>
  </div>
  <div
    class="nhsuk-navigation-container"
  >
    <nav
      aria-label="Primary navigation"
      class="nhsuk-navigation js-enabled"
      role="navigation"
    >
      <ul
        class="nhsuk-header__navigation-list"
      >
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            Home
          </a>
        </li>
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            Get involved
          </a>
        </li>
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            About us
          </a>
        </li>
        <li
          class="nhsuk-header__navigation-item"
        >
          <a
            class="nhsuk-header__navigation-link"
            href="#"
          >
            About us
          </a>
        </li>
        <li
          class="nhsuk-mobile-menu-container nhsuk-mobile-menu-container--visible"
        >
          <button
            aria-expanded="false"
            class="nhsuk-header__menu-toggle nhsuk-header__navigation-link nhsuk-header__menu-toggle--visible"
          >
            <span
              class="nhsuk-u-visually-hidden"
            >
              Browse
            </span>
            More
            <svg
              aria-hidden="true"
              class="nhsuk-icon nhsuk-icon__chevron-down"
              focusable="false"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.5 12a1 1 0 0 1-.29.71l-5 5a1 1 0 0 1-1.42-1.42l4.3-4.29-4.3-4.29a1 1 0 0 1 1.42-1.42l5 5a1 1 0 0 1 .29.71z"
              />
            </svg>
          </button>
          <ul
            class="nhsuk-header__drop-down nhsuk-header__drop-down--hidden"
          >
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                Home
              </a>
            </li>
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                Get involved
              </a>
            </li>
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                About us
              </a>
            </li>
            <li
              class="nhsuk-header__navigation-item"
            >
              <a
                class="nhsuk-header__navigation-link"
                href="#"
              >
                About us
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
